### PR TITLE
Remove hidden fields from the schema

### LINKED ISSUE #2131: camelCase content-type names break content-manager and creation page
**Informations**
- **Node.js version**: v8.11.4
- **npm version**: 5.6.0
- **Strapi version**: 3.0.0-alpha.14.3
- **Database**: mongodb
- **Operating system**: Ubuntu 18.04.1 LTS

**What is the current behavior?**
Adding a content type with a camel cased name (e.g. containing an uppercase letter) breaks the content manager and crud editor.

**Steps to reproduce the problem**
- create a product named `sampleContent`, add a string field and save it.
- try to edit it again, a blank page will appear: http://localhost:1337/admin/plugins/content-manager/ctm-configurations/samplecontent.
- also trying to insert a new document in the db won't work, the creation page will display the loading spinner forever  ( http://localhost:1337/admin/plugins/content-manager/samplecontent/create?redirectUrl=/plugins/content-manager/samplecontent?_limit=10&_page=1&_sort=_id&source=content-manager )

**What is the expected behavior?**

Everything should work as with lowercased collections.

Here's the content-manager blank page stacktrace.

```
TypeError: this.layout is undefined main.js:66
	value main.js:66
	value/< main.js:66
	../strapi-admin/node_modules/strapi-helper-plugin/node_modules/immutable/dist/immutable.js/</e.exports</We.prototype.__iterate main.js:6
	forEach main.js:6
	value main.js:66
	e main.js:66
	o/< main.js:60
	Me main.js:6
	Me main.js:6
	Me main.js:6
	Me main.js:6
	Me main.js:6
	Me main.js:6
	../strapi-admin/node_modules/strapi-helper-plugin/node_modules/immutable/dist/immutable.js/</e.exports</de.prototype.updateIn main.js:6
	o main.js:60
	./node_modules/strapi-helper-plugin/node_modules/redux-immutable/dist/combineReducers.js/t.default/</</< main.js:76
	forEach self-hosted:268:13
	./node_modules/strapi-helper-plugin/node_modules/redux-immutable/dist/combineReducers.js/t.default/</< main.js:76
	../strapi-admin/node_modules/strapi-helper-plugin/node_modules/immutable/dist/immutable.js/</e.exports</de.prototype.withMutations main.js:6
	./node_modules/strapi-helper-plugin/node_modules/redux-immutable/dist/combineReducers.js/t.default/< main.js:76
	c main.js:76
	r/</</< main.js:76
	e/</< main.js:76
	r/< main.js:76
	value main.js:60
	tn main.js:16
	en main.js:16
	Qo main.js:16
	sn main.js:16
	Be main.js:16
	Be self-hosted:1031:17
```

### LINKED ISSUE #2315: Field types are not updated in UI
Informations
* Node.js version: 11.1.0
* npm version: 6.4.1
* Strapi version: 3.0.0-alpha.14.5
* Database: Postgres
* Operating system: macOS Mojave

**What is the current behavior?**

I added a field with the name “isOnline” to an existing type. Then I edited the field directly in the api Model.settings.json and changed its type to boolean and the default to false. On restart, the field is correctly changed in the database and the Content Type Builder changed the type to Boolean as well. It is only the UI in the Content Manager that has not registered the change. The field “isOnline” in the edit view is still displayed as an Integer.

**What is the expected behavior?**

I expect the field to be correctly displayed as a boolean.

I am aware that there is the option to change the field type through the UI with the Content Type Builder. However, there are scenarios where using one is simply not possible. For example when collaborating with several developers using git. In that case the only changes that are being committed are the changed files. Similar if you deploy your changes to heroku. I am aware that there is the option to change the field type through the UI with the Content Type Builder. However, there are scenarios where using one is simply not possible. For example when collaborating with several developers using git. In that case the only changes that are being committed are the changed files. Similar if you deploy your changes to heroku.

### LINKED ISSUE #2333: Clicking around Content-Types eventually yields Javascript error
node 10.13.0
npm 6.4.1
strapi 3.0.0-alpha.14.5

I click on my on Content Type in the Content Type menu on the left and then on Users and I get an error (this does not happen if I click on Users first).  The error is impossible to debug because the javascript is minified. How do develop on a version of strapi that makes it easier to debug?

main.js:16 TypeError: Cannot read property 'toString' of undefined
    at t.value (main.js:48)
    at main.js:48
    at Array.map (<anonymous>)
    at t.n.renderCells (main.js:48)
    at t.value (main.js:48)
    at Mo (main.js:16)
    at So (main.js:16)
    at Ao (main.js:16)
    at Zo (main.js:16)
    at Xo (main.js:16)
zo @ main.js:16
Go.o.callback @ main.js:16
Jt @ main.js:16
$t @ main.js:16
vn @ main.js:16
yn @ main.js:16
hn @ main.js:16
wn @ main.js:16
ze @ main.js:16
main.js:16 Uncaught TypeError: Cannot read property 'toString' of undefined
    at t.value (main.js:48)
    at main.js:48
    at Array.map (<anonymous>)
    at t.n.renderCells (main.js:48)
    at t.value (main.js:48)
    at Mo (main.js:16)
    at So (main.js:16)
    at Ao (main.js:16)
    at Zo (main.js:16)
    at Xo (main.js:16)

### LINKED ISSUE #2355: Content type is missed in content manager sometimes
**Informations**
- **Node.js version**: 10.10.0
- **npm version**: 6.4.1
- **Strapi version**: 3.0.0-alpha.14.5
- **Database**: mongodb
- **Operating system**: mac OS

**What is the current behavior?**
Sometimes, content types are missing in content manager.

**Steps to reproduce the problem**
Try to create a few content types and check if all of them appear in content manager.
This may be hard to reproduce since it is related to the order of models loaded.

**What is the expected behavior?**
All content types should appear in content manager.

**Suggested solutions**
After adding some logs to the code, I found that the problem comes from `getModels` method:
https://github.com/strapi/strapi/blob/eeca356060e6754c14be4baf89e317f7015264ed/packages/strapi-plugin-content-manager/admin/src/containers/SettingsPage/index.js#L40-L58

In my case, one of the content types is iterated before `plugins` and is accumulated. After that the next content type is `plugins`, so `getModels` is called recursively. After returning from recursive calls, previous accumulated value is replaced by the return value, leading to missing content type in content manager.

Concatenating the retuned result to previous accumulated value solves the problem.

### PR DESCRIPTION
**Pull request by soupette**
My PR is a:

Bug fix #2315 fix #2131 fix #2333 
 💅 Enhancement

Main update on the:

Plugin: content-manager

it also fixes #2355 but I like the code from #2356 better

This PR hide the fields that shouldn't be shown in the content manager

### CODE REVIEW

**Comment by lauriejim** [comment]
@soupette does it fix an existing issue ?

**Comment by derrickmehaffy** [comment]
@lauriejim https://github.com/strapi/strapi/issues/2310

**Comment by soupette** [comment]
@derrickmehaffy it fixes a lot of issues...

**Comment by derrickmehaffy** [comment]
@soupette I know but he was wondering if there was an issue attached to it or just a random fix :stuck_out_tongue_closed_eyes:

**Comment by z-ax** [comment]
Wow, what a typo in code!

File: packages/strapi-plugin-content-manager/config/functions/bootstrap.js [hunk 1]
```diff
@@ -51,6 +51,12 @@ module.exports = async cb => {
 
     return acc;
   }, {});
+  // Reference all current models
+  const appModels = Object.keys(pluginsModel).reduce((acc, curr) => {
+    const models = Object.keys(_.get(pluginModels, [curr, 'models'], {}));
```
Comment (henrych4) [inline packages/strapi-plugin-content-manager/config/functions/bootstrap.js]: Is it a typo?
```suggestion
    const models = Object.keys(_.get(pluginsModel, [curr, 'models'], {}));
```

Comment (soupette) [reply packages/strapi-plugin-content-manager/config/functions/bootstrap.js]: You're right this is a typo

**Pull request merged**